DataScience/ProbabilisticParser/parser.py

```diff
@@ -85,3 +85,65 @@ def tag(raw_string):
         tagged[token] = component
 
     return tagged
+
+
+def test(raw_string='ONS LIMITED FLAT 1 12 OXFORD STREET STREET ST1 2FW', verbose=False):
+    """
+    A simple test to check that the calling mechanism from Python gives the same
+    results as if CRFsuite were called directly from the command line. Requires
+    a compiled version of the CRFsuite.
+
+    :param raw_string: input string to test
+    :type raw_string: str
+    :param verbose: additional debugging output
+    :type verbose: bool
+
+    :return: None
+    """
+    print('Input string:', raw_string)
+    print('Python Results:', tag(raw_string))
+
+    tokens = tok.tokenize(raw_string)
+    features = tok.tokens2features(tokens)
+
+    if verbose:
+        print('features:', features)
+
+    tags = TAGGER.tag(features)
+    print('Inferred tags:', tags)
+
+    print('Probability of the sequence:', round(TAGGER.probability(tags), 6))
+    assert round(TAGGER.probability(tags), 6) == 0.956658, 'Sequence probability not correct'
+
+    results = [0.999988, 0.999987, 0.998692, 0.956739, 0.999973, 0.999999, 0.999963, 0.999990, 1., 1.]
+    for i, tg in enumerate(tags):
+        prob = round(TAGGER.marginal(tg, i), 6)
+        print('Marginal probability of', tg, 'in position', i, 'is', prob)
+        assert prob == results[i], 'Marginal Probability of a Label not correct'
+
+    if verbose:
+        print(TAGGER.info().transitions)
+        print(TAGGER.info().state_features)
+        print(TAGGER.info().attributes)
+
+    # store the ItemSequence temporarily
+    tmp = pycrfsuite.ItemSequence(features)
+
+    # write to a text file
+    fh = open('training/test.txt', 'w')
```

Comment:
heh, ok, maybe tmp is not a great name... fh is a standard in python for file handler